### PR TITLE
🎇 Improve local images

### PR DESCRIPTION
**Pull request by rowanc1**
This adds support in myst for local images and attachments.

### CODE REVIEW

File: src/myst.ts [hunk 1]
```diff
@@ -98,7 +99,16 @@ export function parseContent(notebook: StaticNotebook): void {
...
+  cells.forEach(async (cell, index) => {
+    try {
+      // Go through all links and replace the source if they are local
+      await imageUrlSourceTransform(mdast.children[index] as any, {
+        // We must use the cell here rather than the notebook to get the attachments working.
+        parent: cell as any
+      });
+    } catch (error) {
+      // pass
+    }
```
Comment (rowanc1) [inline src/myst.ts]: Looping through all images, and resolving source before rendering.

File: src/images.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  const images = selectAll('image', tree) as Image[];
+  await Promise.all(
+    images.map(async image => {
+      if (!image || !image.url) return;
+      // TODO: not sure why, but the cell seems to have a private _rendermime?
+      // How else to get `attachment:` to work?
+      const rendermime =
+        opts.parent.rendermime ??
+        (opts.parent as any)._rendermime ??
+        (opts.parent.parent as any).rendermime;
```
Comment (rowanc1) [inline src/images.ts]: I could use some help understanding this, the cell seems to have the information for the rendermime image replacement, but that is private (not sure why!?).

**Review by rowanc1 — COMMENTED** [review]
Added a new source transformer based on pointers in #8. For some reason the cell rendermime is private, should we be doing something else?

**Review by fwkoch — APPROVED** [review]
Seems ok, better than it was before. It does feel like we aren't doing this the intended way since `_rendermime` is private...? But the correct way is unclear. Maybe constructing our own `AttachmentsResolver` as is done here? https://github.com/jupyterlab/jupyterlab/blob/6d208970f0b950a1c75c0c08960c651b32a3cc0b/packages/cells/src/widget.ts#L1790

For now, though, this is good.

**Comment by rowanc1** [comment]
Nice -- updated to use the `AttachmentsResolver` directly! Much nicer!!

**Pull request merged**